src/python/tests/test_local_functions.py

```diff
@@ -100,19 +92,19 @@ def failure_str(my_pars, my_units, my_in, my_out, comp_out, field):
     msg += f"stored output: {comp_out[field]}"
     return msg
 
-class LocalFunctionsTest(TestCase):
+class TestLocalFunctions:
     """
     Tests for the local functions.
     """
 
     n_input_sets = 10
     seed = 21062024
     digits = 12
-    test_file = os.path.join(test_answers_dir, "local_function_tests.json")
+    test_file_basename = "local_function_tests.json"
```

Comment:
Why are we no longer storing the json file in the answer test directory?

Reply:
We do still do put the json file in the answer-test-directory. The line that you highlighted has changed to now just specifies the basename of the test-file.

On line 192 we infer the full path to the test-file with
```python
    test_file = os.path.join(answertestspec.answer_dir, self.test_file_basename)
```
and I replaced all references to `self.test_file` to now access `test_file`.

The key change here is that I removed the global `test_answers_dir` variable. Now, the answer test directory is encoded in the value returned by the `answertestspec` fixture. (It is accessed via `answertestspec.answer_dir`).

Would it help if I added an explanatory comment just before the line that you highlighted?

Reply:
Ah, thanks for clarifying. The issue was mine. I think I transposed `test_file` with `test_file_basename` in my head and could no longer find the corresponding instance of `os.path.join`.